docs: add a blog post for MongoDB (#735)
bytebase/bytebase.com: content/blog/bytebase-support-for-mongodb-is-available.md

---

> title: Bytebase Support for MongoDB is Available

Author: > Introducing MongoDB Support in Bytebase

What is the main difference between them?

> title: Bytebase Support for MongoDB is Available

Reviewer: imperative sentence is usually more terse and suitable for the title (because we have limited space) .

Also the current wording leads to bad layout

![CleanShot 2023-01-12 at 00 53 11](https://user-images.githubusercontent.com/230323/211867457-00468c07-979e-492f-a518-1eb0bb9312d5.png)

author: Candy
published_at: 2023/01/12 13:37:17
feature_image: /static/blog/bytebase-support-for-mongodb-is-available/cover.webp
tags: Announcement
featured: true

> description: Bytebase v1.11.0 brings support for MongoDB. This blog post walks you through managing MongoDB with Bytebase. 

Reviewer: ```suggestion
description: Bytebase 1.11.0 brings MongoDB support. This blog post walks you through managing MongoDB with Bytebase. 
```

Author: I'd like to know why Bytebase v1.11.0 is incorrect. I think they are both correct.

Reviewer: They are both correct. Just for consistency, we don't add v in other places.

As to why we choose 1.11.0 over v1.11.0, because less is usually causing less consistency.

---

> ## Why Add MongoDB Support

Reviewer: ```suggestion
## Why MongoDB
```

MongoDB is a popular document-oriented database with the scalability and flexibility that you want with the querying and indexing that you need. MongoDB stores data records as documents (specifically BSON documents) which are gathered together in collections. A database stores one or more collections of documents.

> As the ranking by DBMS’s current popularity from [DB-Engines](https://db-engines.com/en/ranking) shown, until Jan 2023, MongoDB is the only NoSQL database in the top 5 of the overall DB-Engines ranking.
> ![db-rank-3](/static/blog/bytebase-support-for-mongodb-is-available/db-rank-3.webp)

Reviewer: ```suggestion
As the [DB-Engines](https://db-engines.com/en/ranking) ranking shows, MongoDB is the only NoSQL database in the top 5.
![db-rank-3](/static/blog/bytebase-support-for-mongodb-is-available/db-rank-3.webp)
```

> What’s more, our customers say that they wanted to use Bytebase to manage MongoDB, just like they manage database changes to MySQL and PostgreSQL with Bytebase. As you know, Bytebase constantly listens to customers to deliver what they need. So, we support connecting to MongoDB and managing their database changes.

Reviewer: ```suggestion
Some Bytebase customers also have MongoDB besides MySQL/PostgreSQL, and they ask us if Bytebase can support MongoDB so they can manage all transactional database changes via Bytebase. Before MongoDB, Bytebase only supports databases having SQL interface. But the customers' request is reasonable, so the Bytebase team accepts the challenge and today we are excited to announce the MongoDB support.
```

> ## How does it work

Reviewer: ```suggestion
## Manage MongoDB Changes
```

> With Bytebase v1.11.0, you can add an instance to connect to a MongoDB deployment, then you can: 

Reviewer: ```suggestion
With [Bytebase 1.11.0](https://www.bytebase.com/changelog/bytebase-1-11-0), you can add an instance to connect to a MongoDB deployment, then you can: 
```

- Create an issue to create, update and delete documents in UI workflow and GitOps workflow. See the screenshot below to insert documents into the collection “student”.
![insert-doc](/static/blog/bytebase-support-for-mongodb-is-available/insert-doc.webp)

- Explore the migration history in the database details page.
![migration-his](/static/blog/bytebase-support-for-mongodb-is-available/migration-his.webp)

- Run mongosh commands in SQL Editor admin mode.
![admin-mode](/static/blog/bytebase-support-for-mongodb-is-available/admin-mode.webp)

> ## Want to try it Out

Reviewer: ```suggestion
## Try MongoDB Yourself
```

> Bytebase supports MongoDB 4.0, 5.0, and 6.0. Please try it out and find that it's missing something or stumbled upon a bug,  do not hesitate to let us know. We invite you to join our [discord](https://discord.gg/Fac9nmZ95j)) where you can discuss all things Bytebase, share your feedback, and get help from the community.

Reviewer: ```suggestion
Bytebase supports MongoDB 4.0, 5.0, and 6.0. Please follow [installation guide](/docs/get-started/install/overview) to try it out. You can also join our [discord](https://discord.gg/Fac9nmZ95j)) to discuss all things Bytebase, share your feedback, and get help from the community.
```
